test/checks/aria/aria-prohibited-attr.js

```diff
@@ -138,4 +138,85 @@ describe('aria-prohibited-attr', function () {
     var params = checkSetup('<svg id="target" aria-label="hello world"></svg>');
     assert.isFalse(checkEvaluate.apply(checkContext, params));
   });
+
+  describe('widget ancestor', () => {
+    it('should allow aria-label', () => {
+      var params = checkSetup(`
+        <button>
+          <span>
+            <span id="target" aria-label="hello world"></span>
+          </span>
+        </button>
+      `);
+      assert.isFalse(checkEvaluate.apply(checkContext, params));
+    });
+
+    it('should allow aria-labelledby', () => {
+      var params = checkSetup(`
+        <div id="foo">hello world</div>
+        <button>
+          <span>
+            <span id="target" aria-labelledby="foo"></span>
+          </span>
+        </button>
+      `);
+      assert.isFalse(checkEvaluate.apply(checkContext, params));
+    });
+
+    it('should skip "role=none" roles in between ancestor', () => {
+      var params = checkSetup(`
+        <button>
+          <h1 role="none">
+            <span id="target" aria-label="hello world"></span>
+          </h1>
+        </button>
+      `);
+      assert.isFalse(checkEvaluate.apply(checkContext, params));
+    });
+
+    it('should skip "role=presentation" roles in between ancestor', () => {
+      var params = checkSetup(`
+        <button>
+          <h1 role="presentation">
+            <span id="target" aria-label="hello world"></span>
+          </h1>
+        </button>
```

Comment:
Since button has presentational children, it'd be good to have an example of this with a widget that doesn't have this. I think link is the only one.

```suggestion
        <a href="#">
          <h1 role="presentation">
            <span id="target" aria-label="hello world"></span>
          </h1>
        </a>
```